numpy/lib/function_base.py

```diff
@@ -1299,7 +1299,6 @@ def piecewise(x, condlist, funclist, *args, **kw):
                                    isinstance(condlist[0], ndarray))):
         condlist = [condlist]
     condlist = array(condlist, dtype=bool)
-    n = len(condlist)
     # This is a hack to work around problems with NumPy's
     #  handling of 0-d arrays and boolean indexing with
     #  numpy.bool_ scalars
```

Comment:
I know, but the reason why the hack is there (with avoiding scalars) is not the case anymore. So I think it is quite likely that if you remove that said hack, you will also fix your issue. Could you maybe try?
I appreciate the minimal fix, just hoped to nudge you to leave the code nicer then it was ;).